core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -197,7 +197,7 @@ private static void setupLogging(Settings settings, Environment environment) {
 
     private static Tuple<Settings, Environment> initialSettings(boolean foreground) {
         Terminal terminal = foreground ? Terminal.DEFAULT : null;
-        return InternalSettingsPreparer.prepareSettings(EMPTY_SETTINGS, true, terminal);
+        return InternalSettingsPreparer.prepareSettingsAndEnvironment(EMPTY_SETTINGS, terminal);
```

Comment:
I think the only issue with that right now is the last line of the method, which modifies the log file setting to be absolute, after creating the environment.